src/adapter/src/coord/peek.rs

```diff
@@ -308,26 +306,15 @@ impl<S: Append + 'static> crate::coord::Coordinator<S> {
 
         // If the dataflow optimizes to a constant expression, we can immediately return the result.
         if let PeekPlan::FastPath(FastPathPlan::Constant(rows, _)) = fast_path {
-            let mut rows = match rows {
+            let rows = match rows {
                 Ok(rows) => rows,
                 Err(e) => return Err(e.into()),
             };
-            // retain exactly those updates less or equal to `timestamp`.
-            for (_, time, diff) in rows.iter_mut() {
-                use timely::PartialOrder;
-                if time.less_equal(&timestamp) {
-                    // clobber the timestamp, so consolidation occurs.
-                    *time = timestamp.clone();
-                } else {
-                    // zero the difference, to prevent a contribution.
-                    *diff = 0;
-                }
-            }
             // Consolidate down the results to get correct totals.
-            differential_dataflow::consolidation::consolidate_updates(&mut rows);
+            let rows = consolidate_constant_updates(rows);
```

Comment:
Do we need to consolidate constant plans? It'd only ever do something if our own evaluation produces retracted rows, which I don't think is possible?

Reply:
Intuitively I want to agree with you, but I have no idea. Theoretically it's possible for a `Vec<(Row, DIff)>` to have rows that would cancel each other but I have no idea how we'd get there. I'm hoping maybe @frankmcsherry or @aalexandrov know the answer.

Reply:
I don't think we can be sure that `FoldConstants` will always generate consolidated results. For example based on a brief look it seems that at least for the `Union` case we don't do that (see `union_many`), so I suggest to keep this as a precaution if the adapter wants to ensure that the results that are sent back are consolidated.